src/index.js

```diff
@@ -52,64 +27,17 @@ export default function wasm (options = {}) {
           buf[b++] = ((third & 3) << 6) | (table[src.charCodeAt(i+3)] & 63)
         }
 
-        return (${options.load})(buf)
+        return sync ? new WebAssembly.Module(buf) : WebAssembly.compile(buf)
       }
     `.trim(),
     
     transform (code, id) {      
-      if (/\.wasm$/.test(id)) {
-        // Compile wasm -> js
-        return compileWasm(code)
-      } else if (options.compile) {
-        // Compile source code -> wasm -> js
-        return Promise.resolve(options.compile(code, id)).then(compileWasm)
+      if (code && /\.wasm$/.test(id)) {
+        const src = Buffer.from(code, 'binary').toString('base64')
```

Comment:
I don't think this will work in a browser without polyfill - I'd use `btoa` within a browser. See:

https://github.com/ColinEberhardt/rollup-plugin-webassembly/blob/master/src/index.js#L26

Reply:
This is at compile-time inside the Node environment, like [you did here](https://github.com/ColinEberhardt/rollup-plugin-webassembly/blob/master/src/index.js#L21). At runtime it uses the `_wasmLoadModule` function (above this comment) which should work on either platform.  

However, your base64 decoding is different, and looks possibly faster.  Wouldn't mind changing the function to use something similar after experimenting with some benchmarks.

Reply:
Ahhh... my mistake, this is the *encode* not *decode* 👍

Reply:
Side note:  I structured it like this so the decoding can be reused across multiple imports, making the overhead just a tiny function call for aech import.  Versus having each module contain the decoding.